File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
    seen := make(map[int]bool)
    for _, n := range nums {
        seen[n] = true
    }
    longest := 0
    for _, n := range nums {
        if !seen[n-1] {
            seq := 1
            for seen[n+seq] {
                seq++
            }
            longest = max(longest, seq)
        }
    }
    return longest
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        longest = 0
        for n in nums:
            if n-1 not in s:
                seq = 1
                while n+seq in s:
                    seq += 1
                longest = max(longest, seq)    
        return longest